libdistributions_shared.so undefined reference

`make test` now fails for me with:

```
...
Linking CXX executable foo
...
/home/ubuntu/sf/distributions/build/src/libdistributions_shared.so: undefined reference to `google::protobuf::internal::WireFormat::SerializeUnknownFields(google::protobuf::UnknownFieldSet const&, google::protobuf::io::CodedOutputStream*)'
```

Am I missing an undocumented environment variable?

```
$ env | grep DIST
DISTRIBUTIONS_PATH=/home/ubuntu/sf/distributions
DISTRIBUTIONS_USE_PROTOBUF=1
$ env | grep PATH
LD_LIBRARY_PATH=:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib:/home/ubuntu/johann/lib:/home/ubuntu/pomagma/lib
PATH=/home/ubuntu/.virtualenvs/sf/bin:/usr/local/heroku/bin:/home/ubuntu/.virtualenvs/sf/bin:/usr/local/heroku/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/home/ubuntu/bin:/home/ubuntu/tool_config/bin:/home/ubuntu/johann/bin:/home/ubuntu/kazoo/bin:/home/ubuntu/jtext/bin:/home/ubuntu/bin:/home/ubuntu/tool_config/bin:/home/ubuntu/johann/bin:/home/ubuntu/kazoo/bin:/home/ubuntu/jtext/bin
DISTRIBUTIONS_PATH=/home/ubuntu/sf/distributions
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,8 +52,10 @@ endif()
 
 if (DEFINED ENV{DISTRIBUTIONS_USE_PROTOBUF})
   find_package(Protobuf)
-endif()
-if(PROTOBUF_FOUND)
+  if(NOT PROTOBUF_FOUND)
+    message(FATAL_ERROR
+      "DISTRIBUTIONS_USE_PROTOBUF specified but protobuf not found")
+  endif()
   message(STATUS "Using Google Protocol Buffers")
   set(DISTRIBUTIONS_SHARED_LIBS ${DISTRIBUTIONS_SHARED_LIBS} protobuf)
   set(DISTRIBUTIONS_STATIC_LIBS ${DISTRIBUTIONS_STATIC_LIBS} protobuf)

--- a/examples/cmake/CMakeLists.txt
+++ b/examples/cmake/CMakeLists.txt
@@ -5,45 +5,36 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -std=c++0x")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-sign-compare")
 
 
-
 #-----------------------------------------------------------------------------
 # BEGIN generic section
-#
-# To build with local copy of distributions, first build distributions, and
-# then set the environment variable
-# export DISTRIBUTIONS_PATH=/path/to/distributions
-# for example in ~/.bashrc or $VIRTUAL_ENV/bin/postactivate
-
-include(ExternalProject)
-if(DEFINED ENV{DISTRIBUTIONS_PATH})
-  ExternalProject_Add(distributions
-    PREFIX distributions
-    DOWNLOAD_COMMAND ""
-    SOURCE_DIR $ENV{DISTRIBUTIONS_PATH}
-    BINARY_DIR $ENV{DISTRIBUTIONS_PATH}/build
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
-  )
+# germ of a future FindDistributions.cmake
+find_path(DISTRIBUTIONS_INCLUDE_DIR distributions/random.hpp)
+if(NOT DISTRIBUTIONS_INCLUDE_DIR)
+  message(FATAL_ERROR
+    "distributions include dir not found, try setting CMAKE_PREFIX_PATH")
 else()
-  ExternalProject_Add(distributions
-    PREFIX distributions
-    GIT_REPOSITORY https://github.com/forcedotcom/distributions.git
-    INSTALL_COMMAND ""
-  )
+  message(STATUS
+    "using distributions include dir ${DISTRIBUTIONS_INCLUDE_DIR}")
 endif()
-ExternalProject_Get_Property(distributions SOURCE_DIR)
-ExternalProject_Get_Property(distributions BINARY_DIR)
-include_directories("${SOURCE_DIR}/include")
-link_directories("${BINARY_DIR}/src")
-
+find_library(DISTRIBUTIONS_LIBRARIES distributions_shared)
+if(NOT DISTRIBUTIONS_LIBRARIES)
+  message(FATAL_ERROR
+    "distributions libraries not found, try setting CMAKE_PREFIX_PATH")
+endif()
+if(DEFINED ENV{DISTRIBUTIONS_USE_PROTOBUF})
+  find_package(Protobuf)
+  if(NOT PROTOBUF_FOUND)
+    message(FATAL_ERROR
+      "DISTRIBUTIONS_USE_PROTOBUF specified but protobuf not found")
+  endif()
+  set(DISTRIBUTIONS_LIBRARIES ${DISTRIBUTIONS_LIBRARIES} protobuf)
+endif()
 # END generic section
 #-----------------------------------------------------------------------------
 
 
-
 add_executable(foo foo.cc)
-
+include_directories(${DISTRIBUTIONS_INCLUDE_DIR})
 target_link_libraries(foo
-  distributions_shared
+  ${DISTRIBUTIONS_LIBRARIES}
 )

--- a/test_cmake.sh
+++ b/test_cmake.sh
@@ -6,10 +6,5 @@ ROOT=$PWD
 mkdir -p $BUILD
 cd $BUILD
 
-echo '------------'
-echo 'REMOTE BUILD'
+export CMAKE_PREFIX_PATH=$VIRTUAL_ENV
 cmake .. && make && ./foo
-
-echo '-----------'
-echo 'LOCAL BUILD'
-DISTRIBUTIONS_PATH=$ROOT cmake .. && make && ./foo

